emissary-core/src/sam/protocol/datagram/mod.rs

```diff
@@ -111,7 +112,11 @@ impl<R: Runtime> DatagramManager<R> {
                     take::<_, _, ()>(destination.verifying_key().signature_len())(rest)
                         .map_err(|_| Error::InvalidData)?;
 
-                destination.verifying_key().verify(rest, signature)?;
+                if let SigningPublicKey::DsaSha1(_) = destination.verifying_key() {
+                    destination.verifying_key().verify(&sha1::Sha1::digest(rest), signature)?;
+                } else {
+                    destination.verifying_key().verify(rest, signature)?;
+                }
```

Comment:
```suggestion
                match destination.verifying_key() {
                    SigningPublicKey::DsaSha1(_) => return Err(Error::NotSupported),
                    verifying_key => verifying_key.verify(rest, signature)?,
                }
```

FWIW, the spec says it's SHA256 over the payload but I think we should just drop them if nobody's supposed to send them. If for some reason such a datagram is received, it'll be logged as warning in `SamSession`